bridgectrl/merkletree.go

```diff
@@ -113,8 +107,10 @@ func (mt *MerkleTree) initSiblings(ctx context.Context, root []byte, dbTx pgx.Tx
 	return siblings, nil
 }
 
-func (mt *MerkleTree) addLeaf(ctx context.Context, leaf [KeyLen]byte, dbTx pgx.Tx) error {
-	index := mt.count
+func (mt *MerkleTree) addLeaf(ctx context.Context, depositID uint64, leaf [KeyLen]byte, index uint, dbTx pgx.Tx) error {
+	if index != mt.count {
+		return fmt.Errorf("mismatched deposit count: %d, expected: %d", index, mt.count)
```

Comment:
could we detect the missing events with this error?

Reply:
yes